ut/socket_ut.cpp

```diff
@@ -0,0 +1,32 @@
+#include <clickhouse/base/socket.h>
+#include <contrib/gtest/gtest.h>
+#if !defined(_win_)
+#   include <errno.h>
+#   include <fcntl.h>
+#   include <netdb.h>
+#   include <signal.h>
+#   include <unistd.h>
+#   include <arpa/inet.h>
+#endif
+#include <stdio.h>
+#include <string.h>
+#include <iostream>
+#include <fstream>
+#include <thread>
+using namespace clickhouse;
+
+
+
+TEST(Socketcase, connecterror) {
+   int port=9978;
```

Comment:
should I use a random number?

Reply:
I think ideally you should create socket on some port, but don't accept connections on it and try to connect to that socket. Otherwise, I don't see any meaningful difference between predefined or random numbers.

Reply:
make sense, will update accordingly